### PR TITLE
Metal backend: Enable Voxtral Realtime streaming mode

### PR DESCRIPTION
**Pull request by manuelcandales**
This pull request enables running the Voxtral Realtime streaming mode on Metal backend. It refactors the Voxtral Realtime streaming encoder to maintain convolutional (conv) states as internal buffers within the encoder module, rather than passing them explicitly as inputs and outputs. It also introduces AOTI-friendly streaming attention and KV cache implementations.

**Streaming Encoder Refactor (Python & C++):**
- The streaming encoder (`StreamingAudioEncoderExport` in `model.py`) now manages `conv1_state` and `conv2_state` as internal buffers, removing the need to pass these states as explicit inputs/outputs in both Python and C++ runner code.

**Export and Backend Compatibility:**
- Introduces `StandardEncoderSDPA` and `StandardEncoderRingKVCache` classes for AOTI-compatibility, enabling export-friendly streaming attention and KV caching. The encoder dynamically selects the appropriate implementation based on configuration.
- Adjusts how the causal mask is created to avoid using `.item()` on tensors, which is necessary for AOTI export compatibility.

**Metal Backend AOTI Configuration:**
- Sets `"padding_stride_threshold": float("inf")` in Metal backend options to avoid padding stride.

**Auto-detection Logic for Streaming Mode:**
- Updates shell scripts (`export_model_artifact.sh` and `test_model_e2e.sh`) to auto-enable streaming for both XNNPACK and Metal devices.

### CODE REVIEW

File: .ci/scripts/export_model_artifact.sh [hunk 1]
```diff
@@ -262,8 +262,7 @@ if [ "$MODEL_NAME" = "voxtral_realtime" ]; then
   elif [ "$MODE" = "vr-offline" ]; then
     USE_STREAMING="false"
   elif [ -z "$MODE" ]; then
-    # Auto-detect: XNNPACK uses streaming, others use offline
-    if [ "$DEVICE" = "xnnpack" ]; then
+    if [ "$DEVICE" = "xnnpack" ] || [ "$DEVICE" = "metal" ]; then
```
Comment (mergennachin) [inline .ci/scripts/export_model_artifact.sh]: Can we simplify this logic.

use_streaming is true by default, unless mode is specifically vr-offline

and not have this many branches and nest

File: .ci/scripts/test_model_e2e.sh [hunk 2]
```diff
@@ -278,8 +278,7 @@ case "$MODEL_NAME" in
     elif [ "$MODE" = "vr-offline" ]; then
       USE_STREAMING="false"
     elif [ -z "$MODE" ]; then
-      # Auto-detect: XNNPACK uses streaming, others use offline
-      if [ "$DEVICE" = "xnnpack" ]; then
+      if [ "$DEVICE" = "xnnpack" ] || [ "$DEVICE" = "metal" ]; then
```
Comment (mergennachin) [inline .ci/scripts/test_model_e2e.sh]: same comment

File: examples/models/voxtral_realtime/voxtral_realtime_runner.cpp [hunk 3]
```diff
@@ -360,13 +360,7 @@ StreamingSession::StreamingSession(
...
           static_cast<unsigned long long>(std::time(nullptr))),
       input_embeds_buf_(static_cast<size_t>(runner.dim_)) {
-  // Initialize conv states to zero (matches offline encoder's left-padding).
-  // num_mel_bins=128, conv1_pad_=2 → 128*2 = 256 floats
-  conv1_state_.assign(
-      static_cast<size_t>(runner.num_mel_bins_ * runner.conv1_pad_), 0.0f);
-  // enc_dim_=1280, conv2_pad_=2 → 1280*2 = 2560 floats
-  conv2_state_.assign(
-      static_cast<size_t>(runner.enc_dim_ * runner.conv2_pad_), 0.0f);
+  // Conv states are now maintained internally as buffers in the encoder
```
Comment (mergennachin) [inline examples/models/voxtral_realtime/voxtral_realtime_runner.cpp]: Remove this comment

File: examples/models/voxtral_realtime/voxtral_realtime_runner.cpp [hunk 4]
```diff
@@ -502,24 +484,14 @@ bool StreamingSession::try_process_step() {
       ::executorch::aten::ScalarType::Long);
 
   // --- Run streaming encoder ---
+  // Conv states are now maintained internally as buffers
```
Comment (mergennachin) [inline examples/models/voxtral_realtime/voxtral_realtime_runner.cpp]: Remove this comment

**Review by mergennachin — APPROVED** [review]
See inline comment

File: examples/models/voxtral_realtime/model.py [hunk 5]
```diff
@@ -778,18 +893,31 @@ def __init__(self, model: VoxtralRealtimeModel, max_enc_len: int = 750):
         self.n_heads = config.enc_n_heads
         self.head_dim = config.enc_head_dim
 
+        # Register conv states as buffers (mutable state for streaming)
+        self.register_buffer("conv1_state", torch.zeros(1, config.num_mel_bins, 2))
+        self.register_buffer("conv2_state", torch.zeros(1, config.enc_dim, 2))
```
[the remark this answers is not in the document]
Comment (manuelcandales) [reply examples/models/voxtral_realtime/model.py:898]: addressing in follow-up PR

File: examples/models/voxtral_realtime/model.py [hunk 6]
```diff
@@ -741,11 +793,75 @@ def update(
...
+    matching the encoder's convention. Ring buffer enables unlimited streaming.
+    """
+
+    def __init__(self, window_size: int, n_heads: int, head_dim: int):
+        super().__init__()
+        self.window_size = window_size
+        self.buf_size = window_size * 2
+        cache_shape = (1, self.buf_size, n_heads, head_dim)
+        self.register_buffer("k_cache", torch.zeros(cache_shape))
+        self.register_buffer("v_cache", torch.zeros(cache_shape))
```
[the remark this answers is not in the document]
Comment (manuelcandales) [reply examples/models/voxtral_realtime/model.py:828]: addressing in follow-up PR
